x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/compare_synced_integrations.ts

```diff
@@ -0,0 +1,326 @@
+  ccrCustomAssets: { [key: string]: CustomAssetsData }
+): Promise<RemoteSyncedCustomAssetsRecord | undefined> => {
+  if (!ccrCustomAssets) return;
+
+  const abortController = new AbortController();
+
+  try {
+    const installedPipelines = await getPipeline(esClient, CUSTOM_ASSETS_PREFIX, abortController);
+    const installedComponentTemplates = await getComponentTemplate(
+      esClient,
+      CUSTOM_ASSETS_PREFIX,
+      abortController
+    );
+
+    const componentTemplatesByName = (
+      installedComponentTemplates?.component_templates || []
+    ).reduce((acc, componentTemplate) => {
+      if (componentTemplate?.name) {
+        acc[componentTemplate.name] = componentTemplate.component_template?.template;
+      }
+      return acc;
+    }, {} as Record<string, ClusterComponentTemplateSummary>);
+    const componentTemplates = installedComponentTemplates?.component_templates
+      ? componentTemplatesByName
+      : undefined;
+    const result: RemoteSyncedCustomAssetsRecord = {};
+
+    // compare custom pipelines and custom component templates
+    Object.entries(ccrCustomAssets).forEach(([ccrCustomName, ccrCustomAsset]) => {
+      const res = compareCustomAssets({
+        ccrCustomAsset,
+        ingestPipelines: installedPipelines,
+        componentTemplates,
+      });
+      result[ccrCustomName] = res;
+    });
+    return result;
+  } catch (error) {
+    logger.error('error', error?.message);
+    return { error: error?.message };
+  }
+};
+
+const compareCustomAssets = ({
+  ccrCustomAsset,
+  ingestPipelines,
+  componentTemplates,
+}: {
+  ccrCustomAsset: CustomAssetsData;
+  ingestPipelines?: IngestGetPipelineResponse;
+  componentTemplates?: Record<string, ClusterComponentTemplateSummary>;
+}): RemoteSyncedCustomAssetsStatus => {
+  const result = {
+    name: ccrCustomAsset.name,
+    type: ccrCustomAsset.type,
+    package_name: ccrCustomAsset.package_name,
+    package_version: ccrCustomAsset.package_version,
+  };
+
+  if (ccrCustomAsset.type === 'ingest_pipeline') {
+    if (!ingestPipelines) {
+      if (ccrCustomAsset.is_deleted === true) {
+        return {
+          ...result,
+          sync_status: 'completed' as SyncStatus.COMPLETED,
+        };
+      }
+      return {
+        ...result,
+        sync_status: 'synchronizing' as SyncStatus.SYNCHRONIZING,
+      };
+    }
+
+    const installedPipeline = ingestPipelines[ccrCustomAsset?.name];
+    if (ccrCustomAsset.is_deleted === true && installedPipeline) {
+      return {
+        ...result,
+        sync_status: 'failed' as SyncStatus.FAILED,
+        error: `Asset ${ccrCustomAsset.name} marked for deletion found installed`,
+      };
+    } else if (isEqual(installedPipeline, ccrCustomAsset?.pipeline)) {
+      return {
+        ...result,
+        sync_status: 'completed' as SyncStatus.COMPLETED,
+      };
+    } else if (
+      installedPipeline?.version &&
+      installedPipeline.version < ccrCustomAsset.pipeline.version
```

Comment:
the version comparison should be done before the equality check, so we can skip the equality check if the version is not equal